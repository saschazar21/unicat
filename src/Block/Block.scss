@import '../__styles__/variables';

.wrapper {
  padding: spacing(m);
  margin: 0 auto;
  width: 100%;
  overflow: hidden;

  @include media-query-up(m) {
    padding: spacing(l);
  }

  @include media-query-up(l) {
    padding: spacing(xl);
  }
}

@each $variant in $variants {
  .#{$variant} {
    color: color($variant, true);
    background-color: color($variant);
  }
}
