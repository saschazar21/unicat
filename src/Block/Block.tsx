import React, { ReactNode } from 'react';
import classnames from 'classnames';

import { Variant } from '../__types__/global';

import styles from './Block.scss';

export interface BlockProps {
  /* the Block's content */
  children: ReactNode | ReactNode[];
  /* custom className */
  className?: string;
  /* the Block's variant */
  variant?: Variant;
}

export default (props: BlockProps) => {
  const {
    children,
    className: customClassName,
    variant: customVariant,
  } = props;

  const variant = customVariant || 'default';
  const className = classnames(
    styles.wrapper,
    styles[variant],
    customClassName
  );

  return <section className={className}>{children}</section>;
};
