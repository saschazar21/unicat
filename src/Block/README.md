Renders a colored block:

```jsx
import Heading from '../Heading';

<Block>
  <Heading level="h2">I'm a block</Heading>
  <p>And I'm a little bit of text</p>
</Block>;
```
