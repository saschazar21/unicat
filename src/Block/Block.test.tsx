import React from 'react'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Block from './Block';

describe('Block', () => {
  it('renders', () => {
    const wrapper = shallow(<Block>Text</Block>);

    expect(wrapper.find('.default')).toHaveLength(1);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders a variant', () => {
    const wrapper = shallow(<Block variant="light">Text</Block>);

    expect(wrapper.find('.light')).toHaveLength(1);
    expect(wrapper.find('.default')).toHaveLength(0);
  });
});