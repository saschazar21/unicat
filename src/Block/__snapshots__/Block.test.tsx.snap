// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block renders 1`] = `
<section
  className="wrapper default"
>
  Text
</section>
`;
